### PR TITLE
UTOPIA-671: [Backend] AutoSave Functionality - Introducing updatedBy fields + saveID + 409 when conflict

### PR DESCRIPTION
**Pull request by Bottle7**
# Description

This PR includes the following proposed change(s):

- Updates pia intake entity
- Updates mock pia and base entity data to match entity changes
- Generates new migration to update pia intake database table

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Checked container logs to assert successful migration generation and migration runs

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/backend/src/common/entities/base.entity.ts [hunk 1]
```diff
@@ -38,4 +38,19 @@ export class BaseEntity {
...
+  })
+  updatedByGuid: string;
+
+  @Column({
+    name: 'updated_by_username',
+  })
+  updatedByUsername: string;
+
+  @Column({
+    name: 'save_id',
```
Comment (binggaofw) [inline src/backend/src/common/entities/base.entity.ts:53]: maybe named as saved_id so it can consistent with updated_by_*

**Review by binggaofw — APPROVED** [review]

File: src/backend/src/common/entities/base.entity.ts (hunk 1 above)
Comment (kushal-arora-fw) [reply src/backend/src/common/entities/base.entity.ts:53]: POV: save_id sounds better to me

File: src/backend/src/common/entities/base.entity.ts [hunk 2]
```diff
@@ -38,4 +38,19 @@ export class BaseEntity {
...
+    name: 'updated_by_guid',
+  })
+  updatedByGuid: string;
+
+  @Column({
+    name: 'updated_by_username',
+  })
+  updatedByUsername: string;
+
+  @Column({
```
Comment (kushal-arora-fw) [inline src/backend/src/common/entities/base.entity.ts:52]: This column should be generated [auto increment]

File: src/backend/src/modules/pia-intake/pia-intake.service.ts [hunk 3]
```diff
@@ -59,7 +61,14 @@ export class PiaIntakeService {
     this.validateUserAccess(user, userRoles, existingRecord);
 
     // update the record with the provided keys
-    await this.piaIntakeRepository.update({ id }, { ...updatePiaIntakeDto });
+    await this.piaIntakeRepository.update(
```
Comment (kushal-arora-fw) [inline src/backend/src/modules/pia-intake/pia-intake.service.ts:64]: missing save_id comparison, and 409 exception handling. Are you planning to do that in a followup PR?

**Review by kushal-arora-fw — CHANGES_REQUESTED** [review]

**Comment by kushal-arora-fw** [comment]
^^^ Test breaks as well

File: src/backend/src/modules/pia-intake/pia-intake.service.ts (hunk 3 above)
Comment (Bottle7) [reply src/backend/src/modules/pia-intake/pia-intake.service.ts:64]: Yes, I created this PR for the Update Table ticket.

**Comment by Bottle7** [comment]
> ^^^ Test breaks as well

Updated tests to reflect changes.

**Review by kushal-arora-fw — APPROVED** [review]

File: src/backend/src/common/entities/base.entity.ts [hunk 4]
```diff
@@ -38,4 +38,19 @@ export class BaseEntity {
...
+    name: 'save_id',
+  })
```
Comment (kushal-arora-fw) [inline src/backend/src/common/entities/base.entity.ts:54]:
```suggestion
  @Column({
    name: 'save_id',
    nullable: false,
  })
```

**Review by kushal-arora-fw — DISMISSED** [review]
save_id can be nullable.

Please update migration too

File: src/backend/src/common/entities/base.entity.ts (hunk 4 above)
Comment (kushal-arora-fw) [reply src/backend/src/common/entities/base.entity.ts:54]: Also, adding it to the base_entity will add to all the tables. should be inside pia-intake.entity.ts

**Comment by akroon3r** [comment]
This is in an approved state, but Brandon is on Flex today. Make sure this isn't merged until Kushals requested changes have been made please 🚀

**Comment by Bottle7** [comment]
Thanks for picking this up while I was on flex, Kushal! Also, thank you for the sync up and code overview!

**Comment by Bottle7** [comment]
Are there any further comments or suggestions for this PR? If not, can we merge before EOD?

**Pull request merged**